set.mm

```diff
@@ -531784,10 +531788,71 @@ sides of the biconditional correlate (they are the same), if they exist at
   $}
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  1. Restricted Quantifiers
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c wl-in $.
+  $( Redefine the restricted quantifier context to avoid overloading and syntax
+     check errors in mmj2.  This operator does not require ` x ` and ` B `
+     disjoint. $)
+  wwl-in $a wff A. x wl-in B ph $.  $( Extend wff notation for an improved
+     restricted universal quantification. $)
+
+  ${
+    $d x y $.  $d A y $.  $d ph y $.
+    $( Define an improved restricted universal quantification.  This version
+       does not interpret elementhood verbatim in ` A. x wl-in A ph ` .
+       Assuming a real elemnthood leads to awkward consequences should the
+       class ` A ` depend on ` x ` .  Instead a dummy variable ` y ` , disjoint
+       from all other variables, is introduced to describe an elemnt in ` A ` .
+       The subexpression ` A. x ( x = y -> ph ) ` is ` [ y / x ] ph ` in
+       disguise, see ~ sb6 .  This definition lets ` x ` appear as a formal
+       parameter with no connection to ` A ` , but its appearance in ` ph ` is
+       fully honored.  (Contributed by NM, 19-Aug-1993.)  Isolate x from A.
+       (Revised by Wolf Lammen, 21-May-2023.) $)
+    df-wl-ral $a |- ( A. x wl-in A ph
+                  <-> A. y ( y e. A -> A. x ( x = y -> ph ) ) ) $.
+  $}
+
+  ${
+    $d y x $.  $d y ph $.  $d y A $.
+    $( Lemma for ~ wl-nfcral and wl-nfcralv .  (Contributed by Wolf Lammen,
+       23-May-2023.) $)
+    wl-nfcrallem $p |- ( A. y A. x ( y e. A -> ( x = y -> ph ) )
+                              <-> A. x ( x e. A -> ph ) ) $=
+      ( cv wcel weq wi wal alcom bi2.04 equcom imbi1i bitri albii eleq1w imbi1d
+      equsalvw ) CEDFZBCGZAHHZBICIUACIZBIBEDFZAHZBIUACBJUBUDBUBCBGZSAHZHZCIUDUA
+      UGCUATUFHUGSTAKTUEUFBCLMNOUFUDCBUESUCACBDPQRNON $.
+  $}
+
+  ${
+    $d y x $.  $d y ph $.  $d y A $.
+    $( Restricted quantification allows for a simpler expression, if we can
+       assume all appearences of ` x ` in ` A ` are bounded.  (Contributed by
+       Wolf Lammen, 23-May-2023.) $)
+    wl-nfcral $p |- ( F/_ x A -> ( A. x wl-in A ph
+                              <-> A. x ( x e. A -> ph ) ) ) $=
+      ( vy wnfc cv wcel wi wal wwl-in weq wnf wb 19.21t syl albidv wl-nfcrallem
+      nfcr bicomi df-wl-ral 3bitr4g bicomd ) BCEZBFCGAHBIZABCJZUCDFCGZBDKAHZHBI
+      ZDIZUFUGBIHZDIUDUEUCUHUJDUCUFBLUHUJMBDCRUFUGBNOPUIUDABDCQSABDCTUAUB $.
+  $}
+
+  ${
+    $d y ph $.  $d x y A $.
+    $( Restricted quantification allows for a simpler expression, if we can
+       assume ` x ` and ` A ` are disjoint.  (Contributed by Wolf Lammen,
+       23-May-2023.) $)
+    wl-nfcralv $p |- ( A. x wl-in A ph <-> A. x ( x e. A -> ph ) ) $=
+      ( vy wwl-in cv wcel weq wal df-wl-ral 19.21v albii wl-nfcrallem 3bitr2i
+      wi ) ABCEDFCGZBDHAOZBIOZDIPQOBIZDIBFCGAOBIABDCJSRDPQBKLABDCMN $.
+  $}
```

Comment:
Maybe if this theorem was named for example `dfralv`, this would strongly hint that this is a more common definition.
Also in the comment, you could write e.g. "Alternate definition of the restricted universal quantification, assuming `x` does not appear in `A`.".
And to cap it all, I would point to this theorem from the actual (new) definition `df-ral`.

Reply:
`And to cap it all, I would point to this theorem from the actual (new) definition df-ral.`
Then I would lose the 'mathbox only' property.

Lets start in my mathbox first and develop some fundamental theorems there before they are moved to main in a bunch.

Reply:
You can do that all in your mathbox, sorry I should have used the theorem names with `wl-` to be more clear. I was suggesting to have:
- `wl-nfcralv` named `wl-dfralv`,
- the comment of `wl-dfralv` clearly stating it's an alternate definition,
- `wl-dfralv` referenced prominently from `df-wl-ral`.

Reply:
Actually you have done the first two updates in your last commit, sorry this has crossed my comment.

What do you think about the third one, to make a reference to `wl-dfralv` early in the comment of `df-wl-ral`?
For example:
_Define an improved restricted universal quantification.  See `~ wl-dfralv` for a version which requires `A` and `x` to be disjoint, allowing a shorter statement. This version does not interpret elementhood verbatim in ` A. x wl-in A ph ` . [...]_

Reply:
Crossed again! All good now :-)